docs/src/main/tut/dsl.md

```diff
@@ -221,3 +221,119 @@ Ok(drip).run
 ```
 
 ## Matching and extracting requests
+
+A `Request` is a regular `case class` - you can destructure it to extract its
+values. By extension, you can also `match/case` it with different possible
+destructurings. To build these different destructors, you can make use of the
+DSL.
+
+Most often, you extract the `Request` into a HTTP verb and the path, via the
+`->` object. On the right side, you'll have the HTTP verb, on the other side the
+path. Naturally, `_` is a valid matcher too, so any call to `/api` can be
+blocked:
+
+```tut
+HttpService {
+  case request @ _ -> Root / "api" => Forbidden()
+}
+```
+
+To also block all subcalls `/api/...`, you'll need `/:`, since it matches
+everything, and not just the next element:
+
+```tut
+HttpService {
+  case request @ _ -> "api" /: _ => Forbidden()
+}
+```
+
+For matching some verbs, there's `|`:
+
+```tut
+HttpService {
+  case request @ (GET | POST) -> Root / "api"  => ???
+}
+```
+
+Honorable mention: `~`, for matching file extensions.
+
+```tut
+HttpService {
+  case GET -> Root / "example" ~ "json" => ???
+}
+```
+
+### Handling path parameters
+Path params can be extracted and converted to a specific type but are
+`String`s by default. There are numeric extractors provided in the form
+of `IntVar` and `LongVar`.
+
+```tut:book
+import scalaz.concurrent.Task
+
+def getUserName(userId: Int): Task[String] = ???
+
+val usersService = HttpService {
+  case request @ GET -> Root / "users" / IntVar(userId) =>
+    Ok(getUserName(userId))
+}
+```
+
+If you want to extract a variable of type `T`, you can provide a custom extractor
+object which implements `def unapply(str: String): Option[T]`, similar to the way
+in which `IntVar` does it.
+
+```tut:book
+import java.time.LocalDate
+import scala.util.Try
+import scalaz.concurrent.Task
+
+object LocalDateVar {
+  def unapply(str: String): Option[LocalDate] = {
+    if (!str.isEmpty)
+      Try(LocalDate.parse(str)).toOption
+    else
+      None
+  }
+}
+
+def getTemperatureForecast(date: LocalDate): Task[Double] = ???
+
+val dailyWeatherService = HttpService {
+  case request @ GET -> Root / "weather" / "temperature" / LocalDateVar(localDate) =>
```

Comment:
Good example, but maybe show a request that matches this?

GET("/weather/temperature/2016-11-05")